app/models/picture_set.rb

```diff
@@ -6,63 +6,44 @@ class PictureSet
   COMBINED_SUFFIX = '_combined.jpg'.freeze
   PICTURE_PATH = Rails.root.join('public', 'picture_sets').to_s
 
+  attr_accessor :date, :dir, :animation, :combined, :pictures, :next, :last
+
   class << self
 
     def all
       dirs = Dir.glob(File.join(PICTURE_PATH, "*/*#{ANIMATION_SUFFIX}")).map do |animation|
         File.basename(File.dirname(animation))
       end
-      dirs.sort.reverse.map { |dir| new(dir) }
+      Rails.logger.warn "No picture sets found at: #{PICTURE_PATH}" if dirs.empty?
+      dirs.sort.reverse.map { |dir| PictureSet.new(date: dir) }
```

Comment:
``new`` should be enough then you also don't have the dependency to the class name.